olive/cli/base.py

```diff
@@ -38,132 +38,241 @@ def register_subcommand(parser: ArgumentParser):
-    model_group = sub_parser.add_argument_group("model options")
-    model_group.add_argument(
-        "-m",
-        "--model_name_or_path",
-        type=str,
-        required=True,
-        help=(
-            "The model checkpoint for weights initialization. If using an AzureML Registry model, provide the model"
-            " path as 'registry_name:model_name:version'."
-        ),
-    )
-    model_group.add_argument("--trust_remote_code", action="store_true", help="Trust remote code when loading a model.")
-    model_group.add_argument("-t", "--task", type=str, help="Task for which the model is used.")
-
-
-def is_remote_run(args):
-    return all([args.resource_group, args.workspace_name, args.aml_compute])
-
-
-def update_remote_option(config, args, cli_action, tempdir):
-    if args.resource_group or args.workspace_name or args.aml_compute:
-        if not is_remote_run(args):
-            raise ValueError("resource_group, workspace_name and aml_compute are required for remote workflow run.")
-
-        config["workflow_id"] = f"{cli_action}-{hash_dict(config)}"
-
-        try:
-            subscription_id = json.loads(subprocess.check_output("az account show", shell=True).decode("utf-8"))["id"]
-            print("Using Azure subscription ID: %s", subscription_id)
-
-        except subprocess.CalledProcessError:
-            print(
-                "Error: Unable to retrieve account information. "
-                "Make sure you are logged in to Azure CLI with command `az login`."
-            )
-
-        config["azureml_client"] = {
-            "subscription_id": subscription_id,
-            "resource_group": args.resource_group,
-            "workspace_name": args.workspace_name,
-            "keyvault_name": args.keyvault_name,
-            "default_auth_params": {"exclude_managed_identity_credential": True},
-        }
-
-        conda_file_path = Path(tempdir) / "conda_gpu.yaml"
-        with open(conda_file_path, "w") as f:
-            yaml.dump(CONDA_CONFIG, f)
-
-        config["systems"]["aml_system"] = {
-            "type": "AzureML",
-            "accelerators": [{"device": "GPU", "execution_providers": ["CUDAExecutionProvider"]}],
-            "aml_compute": args.aml_compute,
-            "aml_docker_config": {
-                "base_image": "mcr.microsoft.com/azureml/openmpi4.1.0-cuda11.8-cudnn8-ubuntu22.04",
-                "conda_file_path": str(conda_file_path),
-            },
-            "hf_token": bool(args.keyvault_name),
-        }
-        config["workflow_host"] = "aml_system"
-
-
-# TODO(team): Remove this function once the output structure is refactored
-def get_output_model_number(outputs: Dict) -> int:
-    return sum(len(f.nodes) for f in outputs.values())
+    @staticmethod
+    def _add_dataconfig_options(sub_parser):
+        dataset_group = sub_parser.add_argument_group(
+            "data config options, which mutually exclusive with huggingface dataset options"
```

Comment:
User should not need to provide a config json/yaml on the CLI. See how `finetune` CLI handles data